diff -rupN Waterfox/security/sandbox/linux/SandboxOpenedFiles.cpp Waterfox-fixed/security/sandbox/linux/SandboxOpenedFiles.cpp
--- Waterfox/security/sandbox/linux/SandboxOpenedFiles.cpp	2018-04-09 14:01:19.623899549 +1000
+++ Waterfox-fixed/security/sandbox/linux/SandboxOpenedFiles.cpp	2018-04-22 01:40:21.285220672 +1000
@@ -9,6 +9,7 @@
 #include "mozilla/Move.h"
 #include "SandboxLogging.h"
 
+#include <errno.h>
 #include <fcntl.h>
 #include <unistd.h>
 
