diff --git a/virt/kvm/kvm_main.c b/virt/kvm/kvm_main.c
index 54534de..ce67dd6 100644
--- a/virt/kvm/kvm_main.c
+++ b/virt/kvm/kvm_main.c
@@ -1971,6 +1971,7 @@ void kvm_vcpu_block(struct kvm_vcpu *vcpu)
 
 	start = cur = ktime_get();
 	if (vcpu->halt_poll_ns) {
+		bool solo;
 		ktime_t stop = ktime_add_ns(ktime_get(), vcpu->halt_poll_ns);
 
 		do {
@@ -1982,8 +1983,13 @@ void kvm_vcpu_block(struct kvm_vcpu *vcpu)
 				++vcpu->stat.halt_successful_poll;
 				goto out;
 			}
+
+			preempt_disable();
+			solo = single_task_running();
+			preempt_enable();
+
 			cur = ktime_get();
-		} while (single_task_running() && ktime_before(cur, stop));
+		} while (solo && ktime_before(cur, stop));
 	}
 
 	for (;;) {
