diff -Naur xorg-server-1.19.6/Xext/shm.c xorg-server-1.19.6-patched/Xext/shm.c
--- xorg-server-1.19.6/Xext/shm.c	2017-12-21 07:32:33.000000000 +1100
+++ xorg-server-1.19.6-patched/Xext/shm.c	2018-12-10 11:26:54.074732285 +1100
@@ -1200,10 +1200,10 @@
 #ifdef O_TMPFILE
 	fd = open(SHMDIR, O_TMPFILE|O_RDWR|O_CLOEXEC|O_EXCL, 0666);
 	if (fd >= 0) {
-		ErrorF ("Using O_TMPFILE\n");
+		DebugF ("Using O_TMPFILE\n");
 		return fd;
 	}
-	ErrorF ("Not using O_TMPFILE\n");
+	DebugF ("Not using O_TMPFILE\n");
 #endif
 	fd = mkstemp(template);
 	if (fd < 0)
