diff --git a/src/suffer.cpp b/src/suffer.cpp
index 2f33325723..bb3cef8305 100644
--- a/src/suffer.cpp
+++ b/src/suffer.cpp
@@ -318,61 +318,6 @@ void suffer::while_underwater( Character &you )
 
 void suffer::while_grabbed( Character &you )
 {
-    // If you're grabbed you can't move around effectively, triggering the check
-    // TODO: crowd crush without grabs
-    // Count the number of monsters who can shove you
-    // TODO: expand the search area - use the same logic as GROUP_BASH?
-    map &here = get_map();
-    creature_tracker &creatures = get_creature_tracker();
-    int crowd = 0;
-    int impassable_ter = 0;
-    for( auto&& dest : here.points_in_radius( you.pos(), 1, 0 ) ) { // *NOPAD*
-        const monster *const mon = creatures.creature_at<monster>( dest );
-        if( mon && mon->has_flag( mon_flag_GROUP_BASH ) ) {
-            crowd++;
-            add_msg_debug( debugmode::DF_CHARACTER, "Crowd pressure check: monster %s found, crowd size %d",
-                           mon->name(), crowd );
-        }
-        if( here.impassable( dest ) ) {
-            impassable_ter++;
-        }
-    }
-
-    // if we aren't near two monsters with GROUP_BASH we won't suffocate
-    if( crowd < 2 ) {
-        return;
-    }
-    // Getting crushed against the wall counts as a monster
-    if( impassable_ter ) {
-        you.add_msg_if_player( m_bad, _( "You're crushed against the walls!" ) );
-        crowd += impassable_ter;
-    }
-
-    if( crowd == 2 ) {
-        // only a chance to lose breath at low grab chance, none with only a single zombie
-        you.oxygen -= rng( 0, 1 );
-    } else if( crowd <= 4 ) {
-        you.oxygen -= 1;
-    } else if( crowd <= 6 ) {
-        you.oxygen -= rng( 1, 2 );
-    } else if( crowd <= 8 ) {
-        you.oxygen -= 2;
-    }
-
-    // a few warnings before starting to take damage
-    if( you.oxygen <= 5 ) {
-        you.add_msg_if_player( m_bad, _( "You're suffocating!" ) );
-        if( uistate.distraction_oxygen && you.is_avatar() ) {
-            g->cancel_activity_or_ignore_query( distraction_type::oxygen, _( "You're suffocating!" ) );
-        }
-        // your characters chest is being crushed and you are dying
-        you.apply_damage( nullptr, you.get_random_body_part_of_type( body_part_type::type::torso ), rng( 1,
-                          4 ) );
-    } else if( you.oxygen <= 15 ) {
-        you.add_msg_if_player( m_bad, _( "You can't breathe with all this weight!" ) );
-    } else if( you.oxygen <= 25 ) {
-        you.add_msg_if_player( m_bad, _( "You're having difficulty breathing!" ) );
-    }
 }
 
 void suffer::from_addictions( Character &you )
