diff --git a/src/suffer.cpp b/src/suffer.cpp
index f2348799e5..5eb23fe933 100644
--- a/src/suffer.cpp
+++ b/src/suffer.cpp
@@ -340,70 +340,6 @@ void suffer::while_underwater( Character &you )
 
 void suffer::while_grabbed( Character &you )
 {
-    // If you're grabbed you can't move around effectively, triggering the check
-    // TODO: crowd crush without grabs
-    // Count the number of monsters who can shove you
-    // TODO: expand the search area - use the same logic as GROUP_BASH?
-    map &here = get_map();
-    creature_tracker &creatures = get_creature_tracker();
-    int crowd = 0;
-    int impassable_ter = 0;
-    // This looks scary, but it's really just casting the enum to an integer. So medium size characters == 3.
-    int your_size = static_cast<std::underlying_type_t<creature_size>>( you.get_size() );
-    int crush_grabs_req = your_size - 1;
-    // minimum of 1 grabber required
-    crush_grabs_req = std::max( 1, crush_grabs_req );
-
-    for( auto&& dest : here.points_in_radius( you.pos_bub(), 1, 0 ) ) { // *NOPAD*
-        const monster *const mon = creatures.creature_at<monster>( dest );
-        if( mon && mon->has_flag( mon_flag_GROUP_BASH ) ) {
-            crowd++;
-            add_msg_debug( debugmode::DF_CHARACTER, "Crowd pressure check: monster %s found, crowd size %d",
-                           mon->name(), crowd );
-        }
-        if( here.impassable( dest ) ) {
-            impassable_ter++;
-        }
-    }
-
-    add_msg_debug( debugmode::DF_CHARACTER,
-                   "Crowd pressure sum: character size requires %d grabbers, found %d ", crush_grabs_req, crowd );
-
-    // if we aren't near enough monsters with GROUP_BASH we won't suffocate
-    if( crowd < crush_grabs_req ) {
-        return;
-    }
-    // Getting crushed against the wall counts as a monster
-    if( impassable_ter ) {
-        you.add_msg_if_player( m_bad, _( "You're crushed against the walls!" ) );
-        crowd += impassable_ter;
-    }
-
-    if( crowd == crush_grabs_req ) {
-        // only a chance to lose breath at minimum grabs
-        you.oxygen -= rng( 0, 1 );
-    } else if( crowd <= crush_grabs_req * 2 ) {
-        you.oxygen -= 1;
-    } else if( crowd <= crush_grabs_req * 3 ) {
-        you.oxygen -= rng( 1, 2 );
-    } else if( crowd <= crush_grabs_req * 4 ) {
-        you.oxygen -= 2;
-    }
-
-    // a few warnings before starting to take damage
-    if( you.oxygen <= 5 ) {
-        you.add_msg_if_player( m_bad, _( "You're suffocating!" ) );
-        if( uistate.distraction_oxygen && you.is_avatar() ) {
-            g->cancel_activity_or_ignore_query( distraction_type::oxygen, _( "You're suffocating!" ) );
-        }
-        // your characters chest is being crushed and you are dying
-        you.apply_damage( nullptr, you.get_random_body_part_of_type( body_part_type::type::torso ), rng( 1,
-                          4 ) );
-    } else if( you.oxygen <= 15 ) {
-        you.add_msg_if_player( m_bad, _( "You can't breathe with all this weight!" ) );
-    } else if( you.oxygen <= 25 ) {
-        you.add_msg_if_player( m_bad, _( "You're having difficulty breathing!" ) );
-    }
 }
 
 void suffer::from_addictions( Character &you )
